scipy/stats/tests/test_stats.py

```diff
@@ -7971,3 +7971,63 @@ def test_rename_mode_method(fun, args):
     with pytest.raises(TypeError, match=re.escape(err)):
         fun(*args, method='exact', mode='exact')
 
+
+class TestExpectile:
+    def test_same_as_mean(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=0.5), np.mean(x))
+
+    def test_minimum(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=0), np.amin(x))
+
+    def test_maximum(self):
+        rng = np.random.default_rng(42)
+        x = rng.random(size=20)
+        assert_allclose(stats.expectile(x, alpha=1), np.amax(x))
+
+    @pytest.mark.parametrize("alpha", [0.2, 0.5, 0.8])
```

Comment:
Since behaviour seems to change based on `alpha`, (e.g. <= vs. > in the subadditivity), I'd like to make sure we get the boundary right.
```suggestion
    @pytest.mark.parametrize("alpha", [0.2, 0.49, 0.5, 0.51, 0.8])
```

Reply:
In view of [this](https://github.com/scipy/scipy/pull/17039#discussion_r977537904) the boundary should to be that critical anymore and this can be ignored.

OTOH, in terms of expanding the range of `alpha` being tested, ideally I'd like to have a test that ensures we don't crash the method somewhere (or produce garbage results) for alphas close to 0 & 1. E.g.

```
@pytest.mark.parametrize("eps", [1e-3, 1e-7, 1e-11, 1e-15])
@pytest.mark.parametrize("target", [0, 1])
def test_expectile_extreme_alphas(target, eps):
    alpha = target + eps if target == 0 else target - eps
    ...
```

Reply:
What would be the `assert` statement? Our would it just test to pass without error or warning?

Reply:
I added https://github.com/scipy/scipy/pull/17039/commits/7f18ca6f74bc582c6666c4ff27dc18094eb6ca16. That may be enough (or even more than you intended).

Reply:
This is tested in [`test_monotonicity_in_alpha`](https://github.com/scipy/scipy/pull/17039/commits/7f18ca6f74bc582c6666c4ff27dc18094eb6ca16).